ibis/expr/api.py

```diff
@@ -1869,6 +1866,31 @@ def _timestamp_time(arg):
 # ---------------------------------------------------------------------
 # Time API
 
+def between(arg, lower, upper, tz=None):
```

Comment:
Let's call the time zone argument `timezone` to be consistent with the rest of the place we use it.

We should also name this function something else to avoid having two functions named `between`.